src/main/java/com/zendesk/maxwell/producer/MaxwellKafkaProducer.java

```diff
@@ -112,9 +115,35 @@ public MaxwellKafkaProducer(MaxwellContext context, Properties kafkaProperties,
 		this.inflightMessages = new InflightMessageList();
 	}
 
+	//getTopic updates the topic based on the topic per table format (declare by the --kafka_topic_per_table flag)
+	private void getTopic(RowMap r){
+		String original_topic_per_table = this.context.getKafkaTopicPerTableFormat();
+		if(original_topic_per_table.contains("%{database}") || original_topic_per_table.contains("%{table}")) {
+			//replace the %{database} in kafkaTopicPerTable with the datebase name
+			String db_name = r.getDatabase();
+			String kafkaTopicWithFormat = original_topic_per_table.replaceAll("%\\{database\\}", db_name);
+
+			//replace the %{table} in kafkaTopicPerTable with the table name and set the topic name
+			String table_name = r.getTable();
+			this.topic = kafkaTopicWithFormat.replaceAll("%\\{table\\}", table_name);
+
+			//check if the topic exists
+			List<PartitionInfo> partitions_for_kafka;
+			try {
+				partitions_for_kafka = this.kafka.partitionsFor(this.topic);
```

Comment:
you'll want to extract the logic here into a function -- you'll need to pass the return of `partitionsFor` into the line here: https://github.com/zendesk/maxwell/pull/407/files#diff-5c1924a1e21c60a874772298d057e39eR150